src/components/progress-bar/progress-bar.spec.ts

```diff
@@ -0,0 +1,143 @@
+            expect(progressElement.componentInstance.bufferValue).toBe(0);
+            done();
+          });
+    });
+
+    it('should clamp value and bufferValue between 0 and 100', (done:() => void) => {
+      builder
+          .overrideTemplate(TestApp, '<md-progress-bar></md-progress-bar>')
+          .createAsync(TestApp)
+          .then((fixture) => {
+            fixture.detectChanges();
+            let progressElement = getChildDebugElement(fixture.debugElement, 'md-progress-bar');
+            let progressComponent = progressElement.componentInstance;
+
+            progressComponent.value = 50;
+            expect(progressComponent.value).toBe(50);
+
+            progressComponent.value = 999;
+            expect(progressComponent.value).toBe(100);
+
+            progressComponent.value = -10;
+            expect(progressComponent.value).toBe(0);
+
+            progressComponent.bufferValue = -29;
+            expect(progressComponent.bufferValue).toBe(0);
+
+            progressComponent.bufferValue = 9;
+            expect(progressComponent.bufferValue).toBe(9);
+
+            progressComponent.bufferValue = 1320;
+            expect(progressComponent.bufferValue).toBe(100);
+            done();
+          });
+    });
+
+    it('should return the transform attribute based on the value, bufferValue and mode', (done:() => void) => {
+      builder
+          .overrideTemplate(TestApp, '<md-progress-bar></md-progress-bar>')
+          .createAsync(TestApp)
+          .then((fixture) => {
+            fixture.detectChanges();
+            let progressElement = getChildDebugElement(fixture.debugElement, 'md-progress-bar');
+            let progressComponent = progressElement.componentInstance;
+
+            expect(progressComponent.primaryTransform()).toBe('scaleX(0)');
+            expect(progressComponent.bufferTransform()).toBe(undefined);
+
+            progressComponent.value = 40;
+            expect(progressComponent.primaryTransform()).toBe('scaleX(0.4)');
+            expect(progressComponent.bufferTransform()).toBe(undefined);
+
+            progressComponent.value = 35;
+            progressComponent.bufferValue = 55;
+            expect(progressComponent.primaryTransform()).toBe('scaleX(0.35)');
+            expect(progressComponent.bufferTransform()).toBe(undefined);
+
+            progressComponent.mode = 'buffer';
+            expect(progressComponent.primaryTransform()).toBe('scaleX(0.35)');
+            expect(progressComponent.bufferTransform()).toBe('scaleX(0.55)');
+
+
+            progressComponent.value = 60;
+            progressComponent.bufferValue = 60;
+            expect(progressComponent.primaryTransform()).toBe('scaleX(0.6)');
+            expect(progressComponent.bufferTransform()).toBe('scaleX(0.6)');
+            done();
+          });
+    });
+  });
+}
+
+
+/** Gets a child DebugElement by tag name. */
+function getChildDebugElement(parent: DebugElement, selector: string): DebugElement {
```

Comment:
You shouldn't need this function any more since doing `parent.query(By.css(selector))` is shorter than calling this function.